airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -2039,6 +2039,155 @@ def execute(self, context) -> None:
+        For this to work, the service account making the request must have domain-wide
+        delegation enabled.
+    :type delegate_to: str
+    :param location: The location used for the operation.
+    :type location: str
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    template_fields = (
+        'schema_fields',
+        'dataset_id',
+        'table_id',
+        'project_id',
+        'impersonation_chain',
+    )
+    template_fields_renderers = {"schema_fields": "json"}
+    ui_color = BigQueryUIColors.TABLE.value
+
+    @classmethod
+    def _patch_schema(cls, old_schema: Dict, new_schema: Dict) -> None:
+        """
+        Updates the content of "old_schema" with
+        the fields from new_schema. Makes changes
+        in place and hence has no return value.
+        Works recursively for sub-records.
+        Start by turning the schema list of fields into
+        a dict keyed on the field name for both the old
+        and the new schema.
+
+        :param old_schema: Old schema which is updated in-place
+        :type old_schema: dict
+        :param new_schema: Partial schema definition used to patch old schema
+        :type new_schema: dict
+        """
+        new_fields = {field["name"]: field for field in new_schema["fields"] if "name" in field}
+        old_fields = {field["name"]: field for field in old_schema["fields"]}
+
+        # Iterate over all new fields and update the
+        # old_schema dict
+        for field_key in new_fields.keys():
+            # Check if the field exists in the old_schema, if
+            # so change it
+            if field_key in old_fields:
+                old_field = old_fields[field_key]
+                new_field = new_fields[field_key]
+                # Check if recursion is needed
+                if "fields" in new_field:
+                    cls._patch_schema(old_field, new_field)
+                    del new_field["fields"]
+
+                # Do the update
+                old_field.update(new_field)
+
+            # Field didn't exist, add it as a new field
+            else:
+                old_schema["fields"].append(new_fields[field_key])
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        dataset_id: Optional[str] = None,
+        schema_fields: List[Dict[str, Any]],
```

Comment:
Because your Operator uses this as the new schema, is it better to rename it to something more intuitive? `new_schema_fields`?

Reply:
Wanted to be consistend with naming in other operators, can change it if you think intuitive > consistency